### PR TITLE
ci(jenkins): enable CI Pipeline and JJBB

### PR DESCRIPTION
**Pull request by v1v**
## What

Enable the JJBB and a CI Pipeline to run:

- checkout
- pre-commit validation

There will be some follow-ups to add more stages to the CI Pipeline. See https://github.com/elastic/ecs-dotnet/pull/49

### CODE REVIEW

File: .ci/jobs/defaults.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        - change-request:
+            ignore-target-only-changes: false
+        clean:
+          after: true
+          before: true
+        prune: true
+        shallow-clone: true
+        depth: 3
+        do-not-fetch-tags: true
+        submodule:
```
Comment (cachedout) [inline .ci/jobs/defaults.yml]: Do you think it would be good to add a `reference-repo` here?

Comment (v1v) [reply .ci/jobs/defaults.yml]: Good point!
The repo is not as big as other git repos and the number of CI builds might not be too much, so I'd say we don't need it for now. and if we need it then we need to add this repo to the list of the reference repos supported in the CI workers.

Comment (cachedout) [reply .ci/jobs/defaults.yml]: Works for me. 👍

**Comment by codebrain** [comment]
Thanks all.

**Pull request merged**